core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -187,264 +220,337 @@ class DeleteTopicTest extends QuorumTestHarness {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk"))
+  def testDeletingPartiallyDeletedTopic(quorum: String): Unit = {
     /**
-      * A previous controller could have deleted some partitions of a topic from ZK, but not all partitions, and then crashed.
-      * In that case, the new controller should be able to handle the partially deleted topic, and finish the deletion.
-      */
+     * A previous controller could have deleted some partitions of a topic from ZK, but not all partitions, and then crashed.
+     * In that case, the new controller should be able to handle the partially deleted topic, and finish the deletion.
+     */
 
     val replicaAssignment = Map(0 -> List(0, 1, 2), 1 -> List(0, 1, 2))
     val topic = "test"
-    servers = createTestTopicAndCluster(topic, true, replicaAssignment)
+    createTestTopicAndCluster(topic, replicaAssignment = replicaAssignment)
 
     /**
-      * shutdown all brokers in order to create a partially deleted topic on ZK
-      */
-    servers.foreach(_.shutdown())
+     * shutdown all brokers in order to create a partially deleted topic on ZK
+     */
+    brokers.foreach(_.shutdown())
 
     /**
-      * delete the partition znode at /brokers/topics/test/partition/0
-      * to simulate the case that a previous controller crashed right after deleting the partition znode
-      */
+     * delete the partition znode at /brokers/topics/test/partition/0
+     * to simulate the case that a previous controller crashed right after deleting the partition znode
+     */
     zkClient.deleteRecursive(TopicPartitionZNode.path(new TopicPartition(topic, 0)))
     adminZkClient.deleteTopic(topic)
 
     /**
-      * start up all brokers and verify that topic deletion eventually finishes.
-      */
-    servers.foreach(_.startup())
-    TestUtils.waitUntilTrue(() => servers.exists(_.kafkaController.isActive), "No controller is elected")
-    TestUtils.verifyTopicDeletion(zkClient, topic, 2, servers)
+     * start up all brokers and verify that topic deletion eventually finishes.
+     */
+    brokers.foreach(_.startup())
+    TestUtils.waitUntilTrue(() => brokers.asInstanceOf[Seq[server.KafkaServer]].exists(_.kafkaController.isActive), "No controller is elected")
+    TestUtils.verifyTopicDeletion(zkClient, topic, 2, brokers)
+  }
+
+  protected def securityProtocol: SecurityProtocol = SecurityProtocol.PLAINTEXT
+  protected def listenerName: ListenerName = ListenerName.forSecurityProtocol(securityProtocol)
+
+  private def setUpAdminClient(): Unit = {
+    adminClientConfig ++= TestUtils.securityConfigs(Mode.CLIENT, SecurityProtocol.PLAINTEXT, None,
+      "adminClient", TestUtils.SslCertificateCn, None)
+    adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, TestUtils.bootstrapServers(brokers, listenerName))
+    adminClient = TestUtils.createAdminClient(brokers, listenerName, adminClientConfig)
```

Comment:
Please avoid this pattern. This pattern make code very difficult to read. I suggest returning the admin client. E.g. `def setupAdminClient(): Admin`